packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -80,85 +72,46 @@ export class SwapQuoter {
      * Instantiates a new SwapQuoter instance
      * @param   supportedProvider   The Provider instance you would like to use for interacting with the Ethereum network.
      * @param   orderbook           An object that conforms to Orderbook, see type for definition.
-     * @param   options             Initialization options for the SwapQuoter. See type definition for details.
+     * @param   opts                Initialization opts for the SwapQuoter. See type definition for details.
      *
      * @return  An instance of SwapQuoter
      */
-    constructor(supportedProvider: SupportedProvider, orderbook: Orderbook, options: Partial<SwapQuoterOpts> = {}) {
+    public static async createAsync(supportedProvider: SupportedProvider, orderbook: Orderbook, opts: Partial<SwapQuoterOpts> = {}): Promise<SwapQuoter> {
+        const chain = await LiveChain.createAsync({ provider: providerUtils.standardizeOrThrow(supportedProvider) });
```

Comment:
`LiveChain` will automatically fetch the chainId so we can almost get rid of it from `SwapQuoterOpts`. We can't today because the `ExchangeProxyConsumer` still uses it.